File: src/main/java/net/thedragonteam/armorplus/misc/ThreadCosmeticsFetcher.java
package net.thedragonteam.armorplus.misc;

import net.thedragonteam.thedragonlib.util.LogHelper;

import java.io.IOException;
import java.io.InputStreamReader;
import java.net.URL;
import java.util.Properties;

/**
 * @author Sokratis Fotkatzikis - TheDragonTeam
 */
public class ThreadCosmeticsFetcher extends Thread {

    public ThreadCosmeticsFetcher() {
        this.setName("ArmorPlus cosmetics fetcher");
        this.setDaemon(true);
        this.start();
    }

    @Override
    public void run() {
        LogHelper.info("Fetching cosmetics for people...");
        try {
            URL url = new URL("https://download.nodecdn.net/containers/thedragonteam/armorplus-cosmetics.properties");
            Properties specialProperties = new Properties();
            specialProperties.load(new InputStreamReader(url.openStream()));
            CosmeticsRenderInit.parse(specialProperties);

            LogHelper.info("Fetching cosmetics for people done!");
        } catch (IOException e) {
            LogHelper.error("${ThreadCosmeticsFetcher::class.java} Failed", e);
        }
    }
}

File: src/main/java/net/thedragonteam/armorplus/misc/ThreadCosmeticsFetcher.kt
/*
* Copyright (c) TheDragonTeam 2016-2017.
*/

package net.thedragonteam.armorplus.misc

import net.thedragonteam.thedragonlib.util.LogHelper
import java.io.IOException
import java.io.InputStreamReader
import java.net.URL
import java.util.*

/**
 * @author Sokratis Fotkatzikis - TheDragonTeam
 */
class ThreadCosmeticsFetcher : Thread() {
    init {
        this.name = "ArmorPlus cosmetics fetcher"
        this.isDaemon = true
        this.start()
    }

    override fun run() {
        LogHelper.info("Fetching cosmetics for people...")
        try {
            val url = URL("https://download.nodecdn.net/containers/thedragonteam/armorplus-cosmetics.properties")
            val specialProperties = Properties()
            specialProperties.load(InputStreamReader(url.openStream()))
            CosmeticsRenderInit.parse(specialProperties)

            LogHelper.info("Fetching cosmetics for people done!")
        } catch (e: IOException) {
            LogHelper.error("${ThreadCosmeticsFetcher::class.java} Failed", e)
        }

    }
}